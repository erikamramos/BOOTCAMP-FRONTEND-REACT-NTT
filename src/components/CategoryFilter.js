export const filterByCategory = (products, category) => {
  return category ? products.filter((product) => product.category === category) : products;
};

export const populateCategories = (categories, dropdown) => {
  categories.forEach((category) => {
    const option = document.createElement('option');
    option.value = category.slug;
    option.textContent = category.name;
    dropdown.appendChild(option);
  });
};
