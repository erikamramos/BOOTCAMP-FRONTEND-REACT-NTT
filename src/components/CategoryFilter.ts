import { CategoriesResponse } from '../models/Category';

export const renderCategories = (categories: CategoriesResponse, dropdown: HTMLSelectElement): void => {
  categories.forEach((category) => {
    const option = document.createElement('option');
    option.value = category.slug;
    option.textContent = category.name;
    dropdown.appendChild(option);
  });
};