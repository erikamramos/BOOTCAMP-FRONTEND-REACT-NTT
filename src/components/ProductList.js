export const renderProducts = (products, container) => {
  container.textContent = '';
  products.forEach((product) => {
    const productCard = document.createElement('div');
    productCard.className = 'product-card product-card--item';

    // Imagen
    const img = document.createElement('img');
    img.className = 'product-card__image';
    img.src = product.thumbnail;
    img.alt = product.title;

    // Nombre
    const title = document.createElement('h3');
    title.className = 'product-card__name';
    title.textContent = product.title;

    // Categoría
    const category = document.createElement('span');
    category.className = 'product-card__category';
    category.textContent = product.category;

    // Precio
    const price = document.createElement('p');
    price.className = 'product-card__price';
    price.textContent = `Precio: $${product.price}`;

    // Botón
    const button = document.createElement('button');
    button.className = 'product-card__button';
    button.textContent = 'Agregar al carrito';
    button.addEventListener('click', () => incrementCart());

    const icon = document.createElement('i');
    icon.className = 'icon product-card__button-icon';
    icon.dataset.icon = 'shopping-cart';
    
    // Create Card
    button.append(icon);
    productCard.append(img, category, title, price, button);
    container.appendChild(productCard);
    
  });
};

const incrementCart = (() => {
  let cartCount = 0;
  const cartCounterElement = document.getElementById('cart-count');
  return () => {
    cartCount += 1;
    cartCounterElement.textContent = cartCount;
  };
})(); // por qué un iief?
