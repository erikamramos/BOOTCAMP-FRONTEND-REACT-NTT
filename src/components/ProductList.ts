import { Product } from '../models/Product';
import { createRatingStars } from '../utils/ratingStars';

export const renderProducts = (
  products: Product[],
  container: HTMLElement
): void => {
  container.textContent = '';

  products.forEach((product) => {
    const card = document.createElement('div');
    card.className = 'product-card';

    // Imagen del producto
    const img = document.createElement('img');
    img.className = 'product-card__image';
    img.src = product.thumbnail;
    img.alt = product.title;

    // Título
    const title = document.createElement('h3');
    title.className = 'product-card__name';
    title.textContent = product.title;

    // Categoría
    const category = document.createElement('span');
    category.className = 'product-card__category';
    category.textContent = product.category;

    // Precio
    const price = document.createElement('p');
    price.className = 'product-card__price';
    price.textContent = `$${product.price.toFixed(2)}`;

    // Rating (si está entre 0.0 y 5.0)
    const ratingContainer = createRatingStars(product.rating);
    card.appendChild(ratingContainer);

    // Botón
    const button = document.createElement('button');
    button.className = 'product-card__button';
    button.textContent = 'Add to Cart';
    button.addEventListener('click', () => incrementCart());

    const icon = document.createElement('i');
    icon.className = 'icon product-card__button-icon';
    icon.dataset.icon = 'shopping-cart';

    button.append(icon);
    card.append(img, category, title, ratingContainer, price, button);
    container.appendChild(card);
  });
};

// esto no es un componente
const incrementCart = (() => {
  let cartCount = 0;
  const cartCounterElement = document.getElementById('cart-count') as HTMLElement;

  return () => {
    cartCount += 1;
    cartCounterElement.textContent = cartCount.toString();
  };
})();
