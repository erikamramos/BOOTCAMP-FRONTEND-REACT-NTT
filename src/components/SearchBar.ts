import { Product } from '../models/Product';

export const searchProducts = (products: Product[], query: string): Product[] => {
  return products.filter((product) =>
    product.title.toLowerCase().includes(query.toLowerCase()) ||
    product.description.toLowerCase().includes(query.toLowerCase())
  );
};
