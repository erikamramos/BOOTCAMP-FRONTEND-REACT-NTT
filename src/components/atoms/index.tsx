export { Button } from './Button/Button';
export { Icon } from './Icon/Icon';
export { RatingStars } from './RatingStars/RatingStars';
export { Input } from './Input/Input';
export { Select } from './Select/Select';
