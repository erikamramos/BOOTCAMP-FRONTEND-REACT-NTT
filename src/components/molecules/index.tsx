export { Card } from './Card/Card';
export { FormField } from './FormField/FormField';
export {
  Table,
  TableHeader,
  TableBody,
  TableFooter,
  TableRow,
  TableHead,
  TableCell,
} from './Table/Table';
