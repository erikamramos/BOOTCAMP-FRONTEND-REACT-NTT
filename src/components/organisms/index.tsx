export { NavBar } from './NavBar/NavBar';
export { Footer } from './Footer/Footer';
export { Modal } from './Modal/Modal';
