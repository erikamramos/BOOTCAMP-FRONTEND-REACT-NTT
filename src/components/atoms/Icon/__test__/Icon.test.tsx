import { render, screen } from '@testing-library/react';
import { Icon } from '@/components/atoms';

const renderComponent = (props = { name: 'search' }) => {
  return render(<Icon {...props} />);
};

describe('Icon Component', () => {
  it('renders component', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
    expect(screen.getByLabelText(`icon-search`)).toBeInTheDocument();
  });
});
