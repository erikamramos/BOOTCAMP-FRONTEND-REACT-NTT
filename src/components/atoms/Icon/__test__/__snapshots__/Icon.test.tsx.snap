// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component renders component 1`] = `
<div>
  <svg
    aria-label="icon-search"
    fill="#245558"
    height="24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <use
      href="/src/assets/icons/search.svg#icon"
    />
  </svg>
</div>
`;
